### PR TITLE
bar chart unexpected null value

### LINKED ISSUE #94616: Bar charts: x axis ticks display Null value
### What happened?

If I have Null mapping for data turned on and choose X-axis labels minimum spacing to any value except None, spaces between X-axis ticks are filled with Null value.

Image

### What did you expect to happen?

There should be empty space between X-axis ticks.

### Did this work before?

Don't think so, I was able to reproduce it on different Grafana versions.

### How do we reproduce it?

1. Have a bar chart
2. Set mapping for Special Null value
3. Set X-axis labels minimum spacing to any except None

### Is the bug inside a dashboard panel?

Yes

### Environment (with versions)?

Grafana: 10x & 11x
OS: macos
Browser: firefox

### Grafana platform?

Other

### Datasource(s)?

_No response_

### PR DESCRIPTION
**Pull request by yincongcyincong**
Fixes #94616
image

1. Set mapping for Special Null value
image

2. Set X-axis labels minimum spacing to any except None
image

### CODE REVIEW

**Comment by yincongcyincong** [comment]
my pr is check is x axis or not. if yes, don't execute `getValueMappingResult` function.

File: packages/grafana-data/src/field/displayProcessor.test.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { createTheme } from '../themes/createTheme';
 import { FieldConfig, FieldType } from '../types/dataFrame';
 import { DisplayProcessor, DisplayValue } from '../types/displayValue';
 import { ThresholdsMode } from '../types/thresholds';
-import { MappingType, ValueMapping } from '../types/valueMapping';
+import {MappingType, SpecialValueMatch, ValueMapping} from '../types/valueMapping';
```
Comment (Develer) [inline packages/grafana-data/src/field/displayProcessor.test.ts:6]: Let's keep spaces after and before curly brackets 🙂

File: packages/grafana-data/src/field/displayProcessor.ts [hunk 2]
```diff
@@ -15,6 +15,7 @@ import { getValueMappingResult } from '../utils/valueMappings';
 import { FormattedValue, getValueFormat, isBooleanUnit } from '../valueFormats/valueFormats';
 
 import { getScaleCalculator } from './scale';
+import {map} from "rxjs/operators";
```
Comment (Develer) [inline packages/grafana-data/src/field/displayProcessor.ts:18]: Same here 🙏

**Comment by yincongcyincong** [comment]
@Develer thanks for your review. i formated the code.

**Comment by Develer** [comment]
@yincongcyincong I'm not sure if excluding the x-axis from the calculation is the solution here. There may be cases when I want to match some specific values, like this:
Screenshot 2024-10-16 at 13 00 45

So, technically it works as expected 🤷🏻‍♂️
Maybe I'm missing something, I'd like to hear other opinions, but thank you for a solid effort 🙌🏻

**Comment by yincongcyincong** [comment]
@Develer thanks for your reply. how about x axis only can't be affected by `special null value`.

**Comment by leeoniya** [comment]
hi @yincongcyincong, thanks for your PR, it's a very thoughtful attempt, but unfortunately touches a core formatting function that is used all over the codebase to fix one specific issue for one panel.

there is actually a much easier fix here, which is to skip formatting null-valued axis ticks: https://github.com/grafana/grafana/pull/94935. please let us know if that fixes the issue for you.

**Comment by yincongcyincong** [comment]
> hi @yincongcyincong, thanks for your PR, it's a very thoughtful attempt, but unfortunately touches a core formatting function that is used all over the codebase to fix one specific issue for one panel.
> 
> there is actually a much easier fix here, which is to skip formatting null-valued axis ticks: #94935. please let us know if that fixes the issue for you.

thanks for your reply!

**Pull request closed without merging**